### ISSUE TITLE
Create a more specified zone

### ISSUE THREAD

**Issue by vdhicts** [exonet/powerdns-php#21]
The purpose is to create a zone, with more specific options.

## Detailed description

To create a zone, I think it would be nice to supply the Zone resource, instead of the parameters. This way it's easier to create a zone with more specific options, like a `soaEditApi` value.

## Context

This makes it easier to create a more specific zone.

## Possible implementation

```
$zone = new Resources\Zone();
$zone->set...();
...
$powerDns->createZoneFromResource($zone);
```

What needs to be done:
1. Create a `toArray` function on the resource.
2. Create a method which passes the data to `$this->connector->patch()`

**Comment by trizz** [comment]
Hi @vdhicts,

Thanks for the suggestion! I'll add it to our backlog so we can add it. I expect to release it in a new version before may, but I can't promise anything. Of course a PR is also welcome!

**Comment by vdhicts** [comment]
Thank you! If I can find some time, I will give it a shot.

**Comment by trizz** [comment]
@vdhicts We've just released version [2.2.0](https://github.com/exonet/powerdns-php/releases/tag/v2.2.0) with this feature. See [the new example](https://github.com/exonet/powerdns-php/blob/master/examples/new_domain_from_zone_resource.php) on how to use it.

**Comment by vdhicts** [comment]
Very nice, thank you!
